src/ui/components/FogBackground/component.ts

```diff
@@ -1,10 +1,59 @@
 import Component, { tracked } from '@glimmer/component';
+import FloatingParticle from './particle';
 
 export default class FogBackground extends Component {
-  @tracked('args')
-  get opacityStyle(): string {
-    let { intensity } = this.args;
-    let opacity = Math.pow(2.5, intensity) / 10;
-    return `opacity: ${(opacity > 1) ? 1 : opacity}`;
+  particles: FloatingParticle[] = [];
+  ctx: CanvasRenderingContext2D;
+  checkDeleted: boolean = false;
+  fogImage: HTMLImageElement;
+
+  particlesForIntensity(intensity: number): number {
+    return (intensity === 0) ? 0 : 0.10 * Math.pow(2, intensity);
+  }
+
+  didInsertElement() {
+    if (window.requestAnimationFrame) {
+      this.particlesBackground();
+    }
+  }
+
+  async particlesBackground() {
+    let canvas: HTMLCanvasElement = document.querySelector('#ParticlesBackgroundCanvas');
+    let image: HTMLImageElement = document.querySelector('#ParticleImage');
+    let ctx = canvas.getContext('2d');
+
+    canvas.width = window.innerWidth;
+    canvas.height = window.innerHeight;
+
+    this.ctx = ctx;
+    this.fogImage = image;
+    this.addParticles(115);
+    window.requestAnimationFrame(this.drawParticles);
+  }
+
+  addParticles(extras: number) {
```

Comment:
I think this does not actually **add** particles but simply set them up initially? Or is there any accumulative behavior here that I don't see?